charts/kuberpult/tests/charts_test.go

```diff
@@ -0,0 +1,539 @@
+/*This file is part of kuberpult.
+
+Kuberpult is free software: you can redistribute it and/or modify
+it under the terms of the Expat(MIT) License as published by
+the Free Software Foundation.
+
+Kuberpult is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+MIT License for more details.
+
+You should have received a copy of the MIT License
+along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>.
+
+Copyright 2023 freiheit.com*/
+/*
+This file is part of kuberpult.
+
+Kuberpult is free software: you can redistribute it and/or modify
+it under the terms of the Expat(MIT) License as published by
+the Free Software Foundation.
+
+Kuberpult is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+MIT License for more details.
+
+You should have received a copy of the MIT License
+along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>.
+
+Copyright 2023 freiheit.com
+*/
+package main_test
+
+import (
+	"bytes"
+	"fmt"
+	"gopkg.in/yaml.v3"
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	"math/rand"
+	"os"
+	"os/exec"
+	k8sYaml "sigs.k8s.io/yaml" //Needed to properly parse the yaml generated by helm, "gopkg.in/yaml.v3" cannot do it properly
+	"strconv"
+	"strings"
+	"testing"
+)
+
+type ValuesDoc struct { //Auto generated using https://zhwt.github.io/yaml-to-go/. In case you made any changes, post the content of values.yaml there and replace this struct
```

Comment:
Interesting approach, but I don't want to rely on external tools.
If you have generated code, then it needs to be part of the pipeline and generated automatically. But actually I don't think we need generated code here.

I would suggest to replace this struct and instead just have string that is the `content` of the yaml file. Then you can still easily only set a couple values, don't need to generate code, and the `setup` function is also not required. And we are then also able (in the future) to test for wrong types (string vs bool).